src/adaptors/giveth/index.js

```diff
@@ -0,0 +1,207 @@
+const { request, gql } = require('graphql-request');
+const utils = require('../utils');
+const { default: BigNumber } = require('bignumber.js');
+
+const urlUniswapV2 = 'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v2'
+const urlHoneyswapV2 = 'https://api.thegraph.com/subgraphs/name/1hive/honeyswap-v2';
+const urlGivEconomyMainnet = 'https://api.thegraph.com/subgraphs/name/giveth/giveth-economy-mainnet'
+const urlGivEconomyGnosis = 'https://api.thegraph.com/subgraphs/name/giveth/giveth-economy-xdai';
+
+// GIV Token
+const givTokenMainnetAddress = '0x900db999074d9277c5da2a43f252d74366230da0';
+const givTokenGnosisAddress = '0x4f4f9b8d5b4d0dc10506e5551b0513b61fd59e75';
+
+// GIV 100% UNIPOOL LM
+const givMainnetContractInfo = '0x4b9efae862a1755f7cecb021856d467e86976755';
+const givGnosisContractInfo = '0xd93d3bdba18ebcb3317a57119ea44ed2cf41c2f2';
+
+// Giv - Dai Mainnet LP
+const givDaiPairMainnetAddress = '0xbeba1666c62c65e58770376de332891b09461eeb';
+const givDaiPairUnipoolContractInfo = '0xa4523d703f663615bd41606b46b58deb2f926d98';
+
+// Giv - Xdai Gnosis LP
+const givDaiPairGnosisAddress = '0xb7189a7ea38fa31210a79fe282aec5736ad5fa57';
+const givDaiPairGnosisUnipoolContractInfo = '0x24a6067fed46dc8663794c4d39ec91b074cf85d4';
+
+const tokenPairPoolQuery = gql`
+  {
+    pair(id: "<PLACEHOLDER>") {
+      id
+      reserveUSD
+      volumeUSD
+      reserve0
+      reserve1
+      totalSupply
+      token0 {
+        id
+        symbol
+      }
+      token1 {
+        id
+        symbol
+      }
+    }
+  }
+`;
+
+const unipoolContractInfoQuery = gql`
+  {
+    unipoolContractInfo(id: "<PLACEHOLDER>" ) {
+      id
+      totalSupply
+      rewardRate
+    }
+  }
+`;
+
+const calculatePairApy = async (chainString, entry, contractInfo) => {
+  const givTokenAddress = chainString === 'ethereum' ? givTokenMainnetAddress : givTokenGnosisAddress;
+
+  const tokenReserve = BigNumber(
+    entry.token0.id.toLowerCase() !== givTokenAddress
+      ? entry.reserve1
+      : entry.reserve0,
+  );
+
+  const lp = BigNumber(entry.totalSupply)
+    .times(10 ** 18)
+    .div(2)
+    .div(tokenReserve);
+
+  const totalSupply = BigNumber(contractInfo.totalSupply);
+
+  const apr = totalSupply.isZero()
+    ? null
+    : BigNumber(contractInfo.rewardRate)
+        .div(totalSupply)
+        .times('31536000')
```

Comment:
@CarlosQ96 What is `31536000` ?
I think it's good to have a comment for these numbers

Reply:
Its the seconds in a year! you are right ill add it as a constant.